File: Lab01/Lab01/Bill.cs
namespace Lab01;

internal class Bill
{
    public double Limit { get; set; }
    public double Debt { get; set; }

    public Bill(int limit)
    {
        Limit = limit;
        Debt = 0;
    }

    public bool Check(double amount)
    {
        return Debt + amount < Limit;
    }

    public void ChangeLimit(double amount)
    {
        Console.WriteLine($"Limit was changed from {Limit}$ to {amount}$.");
        Limit = amount;
    }

    public void AddDebt(double amount)
    {
        Debt += amount;
        Console.WriteLine($"{amount}$ added to debt. Current debt: {Debt}.");
    }

    public void Pay(double amount)
    {
        Debt -= amount;
        Console.WriteLine($"{amount}$ subtracted from debt. Current debt: {Debt}.");
    }
}

File: lab 1/bill.py
class Bill:
    def __init__(self, limitingAmount):
        self.limitingAmount = limitingAmount
        self.currentDebt = 0.0

    def check(self, amount):
        return self.currentDebt + amount <= self.limitingAmount

    def add(self, amount):
        if self.check(amount):
            self.currentDebt += amount
        else:
            print("Ліміт перевищено!")

    def pay(self, amount):
        if amount <= self.currentDebt:
            self.currentDebt -= amount
        else:
            self.currentDebt = 0.0

    def changeTheLimit(self, amount):
        self.limitingAmount = amount

    def getLimitingAmount(self):
        return self.limitingAmount

    def getCurrentDebt(self):
        return self.currentDebt
